packages/lib/src/vis/raw/utils.ts

```diff
@@ -0,0 +1,11 @@
+import { assertDefined, assertNonNull } from '@h5web/shared/guards';
+import type { TypedArray } from 'ndarray';
+
+export function getTypedArrayClass(arr: TypedArray): string {
+  const res = /^\[object (.+)\]$/u.exec(Object.prototype.toString.call(arr));
+  assertNonNull(res);
+
+  const [, arrClass] = res;
+  assertDefined(arrClass);
+  return arrClass;
```

Comment:
Since it is just for formatting perhaps we should return a default value (e.g. `TypedArray`) instead of asserting ?

Reply:
For some reason, I had found a much simpler way to do this but hadn't pushed it: `value.constructor.name` :sweat_smile: